src/olympia/accounts/helpers.py

```diff
@@ -20,6 +47,42 @@ def fxa_config(context):
     return config
 
 
+def link_with_final_destination(context, base):
+    return urlparams(base, to=path_with_query(context['request']))
+
+
+def path_with_query(request):
+    next_path = request.path
+    qs = request.GET.urlencode()
+    if qs:
+        return '{next_path}?{qs}'.format(next_path=next_path, qs=qs)
```

Comment:
The base64 encoding requires a `str`, but IMHO that `path_with_query()` should just return an unicode string, which you'll `.encode('utf-8')` to get a `str` right before doing the base64 encoding.

With that strategy you need the `u`, otherwise:
```
>>> next_path = u'lolé'
>>> qs = {}
>>> 
>>> '{next_path}?{qs}'.format(next_path=next_path, qs=qs)
UnicodeEncodeError: 'ascii' codec can't encode character u'\xe9' in position 3: ordinal not in range(128)
>>> u'{next_path}?{qs}'.format(next_path=next_path, qs=qs)
u'lol\xe9?{}'
```